### PR TITLE
switch from publicsuffix to addr

### PR DESCRIPTION
**Pull request by rushmorem**
Since v2, domain name parsing logic was moved to the addr crate.

### CODE REVIEW

**Comment by JuxhinDB** [comment]
Thanks for taking the time to update this @rushmorem! I'll be going over it properly later today to see if we can go ahead and merge it. :-)

**Comment by rushmorem** [comment]
No problem :)

**Comment by rushmorem** [comment]
No problem :)

**Review by JuxhinDB — APPROVED** [review]

File: twistrs/src/constants.rs [hunk 1]
```diff
@@ -20,20 +19,6 @@ lazy_static! {
...
-    //                over HTTP. In the future, this should be kept lo-
-    //                cally and call List::from_str instead.
-    //
-    //                At first we used List::from_path and pointed to a
-    //                local .dat file containing the TLDs, however giv-
-    //                en that this is a library, this is not the right
-    //                way to go about it.
-    //
-    //  Ref: https://docs.rs/publicsuffix/1.5.4/publicsuffix/struct.List.html
-    pub static ref EFFECTIVE_TLDS: List =
```
[the remark this answers is not in the document]
Comment (rushmorem) [reply twistrs/src/constants.rs:34]: @JuxhinDB By default, yes. By default, `addr` uses the `psl` crate which compiles the list down to native Rust code for maximum performance. That behaviour can be changed to use the `publicsuffix` crate which is dynamic or to allow users to choose any implementation of `psl_types::List` they like.

**Pull request merged**